File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    box = []
    for i in moves:
        for j in board:
            if j[i-1] != 0:
                box.append(j[i-1])
                j[i-1] = 0
                break
        while len(box) > 1 and box[-1] == box[-2]:
            if box[-1] == box[-2]:
                box = box[:-2]
                answer += 2
    
    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Deque<Integer> que = new ArrayDeque<>();
        for (int m : moves) {
            for (int i = 0; i < board.length; i++) {
                if (board[i][m-1] != 0) {
                    que.add(board[i][m-1]);
                    board[i][m-1] = 0;
                    break;
                }
            }
            if (que.size() >= 2) {
                int a = que.removeLast();
                int b = que.removeLast();
                if (a == b) {
                    answer += 2;
                } else {
                    que.add(b);
                    que.add(a);
                }
            }
        }
        return answer;
    }
}